doc/IDSL.md

```diff
@@ -86,32 +86,32 @@ The value type of a dictionary can be any IDSL type. However, the key type of a
 
 IDSL interface is the core of what makes up Robocomp. 
 
-For communication between components to take place, you must invoke operations on an components proxy, which refers to an interface implimented in the component. To make the structure accessible, you must create an interface that allows clients to access the structure. 
+For communication between components to take place, you must invoke operations on a component's proxy, which refers to an interface implemented in the component. To make the structure accessible, you must create an interface that allows clients to access the structure. 
 
-A component may impliment one or more interfaces and each of those interface will have its own unique proxy.
+A component may implement one or more interfaces and each of those interfaces will have its unique proxy.
 
-You can think of an interface definition as the equivalent of the public part of a C++ class definition, or a Java interface, and of operation definitions as (virtual) member functions. 
+You can think of an interface definition as the equivalent of the public part of a C++ class definition, or a Java interface, and operation definitions as (virtual) member functions. 
 
 Note that nothing but operation definitions are allowed to appear inside an interface definition. In particular, you cannot define a type, an exception, or a data member inside an interface.
 
 ### Operations
 
 Operations are simply methods defined in the interface which can be invoked using a proxy to the interface.
 
-An operation definition must contain a return type and parameter definitions (it can also be left empty). You must use void to indicate that an operation returns no value — there is no default return type for operations.
+An operation definition must contain a return type and parameter definitions (it can also be left empty). You must use void to indicate that a operation returns no value — there is no default return type for operations.
```

Comment:
```suggestion
An operation definition must contain a return type and parameter definitions (it can also be left empty). You must use void to indicate that an operation returns no value — there is no default return type for operations.
```